src/transport/raw/ActiveConnectionState.h

```diff
@@ -0,0 +1,119 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the CHIP Active Connection object that maintains TCP connections.
+ */
+
+#pragma once
+
+#include <inet/IPAddress.h>
+#include <inet/InetInterface.h>
+#include <inet/TCPEndPoint.h>
+#include <lib/core/CHIPCore.h>
+#include <transport/raw/PeerAddress.h>
+#include <transport/raw/TCPConfig.h>
+
+namespace chip {
+namespace Transport {
+
+/**
+ *  The State of the TCP connection
+ */
+enum class State
+{
+    kNotReady    = 0, /**< State before initialization. */
+    kInitialized = 1, /**< State after class is listening and ready. */
+    kConnecting  = 3, /**< Connection with peer has been initiated. */
+    kConnected   = 4, /**< Connected with peer and ready for Send/Receive. */
+    kClosed      = 5, /**< Connection is closed. */
+};
+
+struct AppConnCallbackCtxt;
```

Comment:
This is a pretty cryptic name, and again just putting in in Transport:: is strange.

Reply:
Changed to AppTCPConnectionCallbackCtxt